Framework/CurveFitting/inc/MantidCurveFitting/GSLJacobian.h

```diff
@@ -127,6 +127,8 @@ class JacobianImpl1 : public API::Jacobian {
   }
   /// overwrite base method
   void set(size_t iY, size_t iP, double value) override {
+    if (iP >= m_index.size())
+      return;
```

Comment:
What is the reasoning behind this change? Why not throw for this out of range index?

Reply:
The crystal field fitting functions can change their number of active fitting parameters during the fit. When it happens after an iteration is finished there is a special exception that signals the minimizer to re-initialize itself. But it can happen during a numeric derivative calculation: a field parameter is incremented, a new peak may appear and it may have a free fitting parameter, so the number of parameters change. You cannot throw and re-initialize the minimizer at this point because the same will happen again. If you ignore these "virtual" extra parameters by the time the derivative calculations finish the number of parameters will return to the original value. If you notice the `get(...)` method doesn't skip any extra indices because if there are any it is an error.

Reply:
Awesome. Can you add this explanation to the code?

Reply:
All of it?

Reply:
Or point to a place where this is explained in a similar way?

Reply:
OK